### PR TITLE
🐬 Chore / Update MySql Integration Tests

### PR DESCRIPTION
**Pull request by demariadaniel**
## Link to Issue

#1204 

## Description

Partial improvements to update `testcontainers` and `mysql` libraries
This branch brings integration tests to 4 passing &  42 failing (majority are fixed in next PR)

~~pt1 Prettier/Organization https://github.com/icgc-argo/argo-clinical/pull/1209/files merged~~
pt2 MySql Changes: https://github.com/icgc-argo/argo-clinical/pull/1213
pt3 Mongo Changes: https://github.com/icgc-argo/argo-clinical/pull/1203
pt4 Final Mongoose Tests Fixed: https://github.com/icgc-argo/argo-clinical/pull/1210

### Type of Change

- [x] Bug
- [x] Refactor
- [ ] New Feature
- [ ] Release Candidate

### Checklist before requesting review:

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -132,7 +132,7 @@
 		"morgan": "^1.10.0",
 		"mquery": ">=3.2.3",
 		"multer": "^1.4.5-lts.1",
-		"mysql": "^2.18.1",
+		"mysql2": "^3.11.3",
```
Comment (demariadaniel) [inline package.json:135]: See: https://www.npmjs.com/package/mysql2#history-and-why-mysql2

File: src/bootstrap.ts [hunk 2]
```diff
@@ -135,15 +135,16 @@ const setJwtPublicKey = (keyUrl: string) => {
...
-		password: conf.password,
-		user: conf.user,
-		port: conf.port,
-		timeout: conf.timeout,
+		database,
+		host,
+		password,
+		user,
+		port,
+		connectTimeout,
```
Comment (demariadaniel) [inline src/bootstrap.ts:147]: MySql2 has Types so there are a few updates to align our code with their library

File: src/bootstrap.ts [hunk 3]
```diff
@@ -156,8 +157,8 @@ const setupRxNormConnection = (conf: RxNormDbConfig) => {
 
 async function pingRxNorm(pool: Pool) {
 	try {
-		const query = promisify(pool.query).bind(pool);
-		await query('select 1');
+		const connection = await pool.getConnection();
+		await connection.query('select 1');
```
Comment (ciaranschutte) [inline src/bootstrap.ts:161]: here `getConnection` is used before querying, later in `MysqlRxNormService` class `query` is used directly from `pool`. Might be good to keep it consistent. What is the difference between these two approaches?

File: test/integration/testutils.ts [hunk 4]
```diff
@@ -173,3 +172,7 @@ export async function findInDb(dbUrl: string, collection: string, filter: any) {
 	await conn.close();
 	return result;
 }
+
+export const RXNORM_DB = 'rxnorm';
+export const RXNORM_USER = 'clinical';
+export const RXNORM_PASS = 'password';
```
Comment (ciaranschutte) [inline test/integration/testutils.ts:178]: if moving these around, may as well pop them into a "constants" file rather than utils

**Review by ciaranschutte — APPROVED** [review]
looks good!

File: src/bootstrap.ts (hunk 3 above)
Comment (demariadaniel) [reply src/bootstrap.ts:161]: edit: actually, pool.query is a source of connection errors in our tests
So I'm going to use getConnection with a manual release

File: src/bootstrap.ts [hunk 5]
```diff
@@ -176,7 +177,11 @@ export const run = async (config: AppConfig) => {
 	}
 
 	// RxNorm Db
-	setupRxNormConnection(config.rxNormDbProperties());
+	try {
+		setupRxNormConnection(config.rxNormDbProperties());
+	} catch (error) {
+		L.error('Error at setupRxNormConnection', error);
+	}
```
[the remark this answers is not in the document]
Comment (demariadaniel) [reply src/bootstrap.ts]: This was added for debugging connection errors and in review can be removed
I looked at the potential for errors with `initPool` and any errors here are handled by code elsewhere in the tests

**Pull request merged**